code/controllers/configuration/config_entry.dm

```diff
@@ -156,44 +170,58 @@
 	config_entry_value = new_list
 	return TRUE
 
-/datum/config_entry/keyed_flag_list
-	abstract_type = /datum/config_entry/keyed_flag_list
+//Format: /datum/config_entry/keyed_list/key_type/value_type
+/datum/config_entry/keyed_list
+	abstract_type = /datum/config_entry/keyed_list
 	config_entry_value = list()
 	dupes_allowed = TRUE
+	var/key_mode
+	var/value_mode
+	var/splitter = " "
 
-/datum/config_entry/keyed_flag_list/ValidateAndSet(str_val)
+/datum/config_entry/keyed_list/ValidateAndSet(str_val)
 	if(!VASProcCallGuard(str_val))
 		return FALSE
-	return ValidateKeyedList(str_val, LIST_MODE_FLAG, " ")
-
-/datum/config_entry/keyed_number_list
-	abstract_type = /datum/config_entry/keyed_number_list
-	config_entry_value = list()
-	dupes_allowed = TRUE
-	var/splitter = " "
+	return ValidateKeyedList(str_val, key_mode, value_mode, splitter)
 
-/datum/config_entry/keyed_number_list/vv_edit_var(var_name, var_value)
+/datum/config_entry/keyed_list/vv_edit_var(var_name, var_value)
 	return var_name != "splitter" && ..()
 
-/datum/config_entry/keyed_number_list/ValidateAndSet(str_val)
-	if(!VASProcCallGuard(str_val))
-		return FALSE
-	return ValidateKeyedList(str_val, LIST_MODE_NUM, splitter)
+/datum/config_entry/keyed_list/text
+	abstract_type = /datum/config_entry/keyed_list/text
+	key_mode = KEY_MODE_TEXT

```

Comment:
With both a key and value type I don't know if it's appropriate anymore to have a different type per mode. This can quickly get out of hand in combinations